### DISCUSSION TITLE
How to aggregate cluster/node info in telemetry report?

### DISCUSSION THREAD

**Discussion by pmalek** [Kong/kubernetes-telemetry#10] [General]
Given [this](https://github.com/Kong/kubernetes-telemetry/blob/f230f4ad2b6078155bd359d626a7a39d36e200f2/keps/0001-telemetry.md#catalog)

> Catalog
>
> When running the telemetry library a number of Jobs will be available for loading via a Catalog. The catalog will effectively just be a map of named jobs to their underlying Executors which can be compiled into a Workflow.
>
> For the first iteration of this tool, we want to include Jobs which can collect the following information for general environment:
>
> - Orchestration Platform
> - Kubernetes Version
> - Feature Flags
> - Connection to a Service Mesh
> - Number of pods
> - Number of services
> - Architecture

We have to somehow think about how to report non-homogenous clusters, e.g. where half of cluster is running k8s nodes version v1.23 and half is v1.24 (or a variation of that).

How would we like to report this? Do we want to report all unique values in a relevant report's tag, for instance, given that `version` is the tag under which we report the cluster/nodes version we could serialize it into a comma separated values like to: `version=1.23,1.24`. This would make it non-trivial to parse, the consumer of this report would have to know how to deserialize it (we would provide the library that would do that but still).

Similarly other node related attributes would have to be treated, e.g. architecture, OS.

**Comment by shaneutt** [comment]
If I'm not mistaken in KIC we currently provide that versioning information based entirely on the `/version` response of the Kubernetes API. With that implementation in mind I believe we're currently blind to clusters running in mixed version mode. Furthermore, I'm not really sure that's something we should fix because the API is our only integration point?

Can you help me to better understand the value proposition of being able to report on these mixed versions?

**Comment by scseanchow** [comment]
> we could serialize it into a comma separated values like to: version=1.23,1.24

> This would make it non-trivial to parse, the consumer of this report would have to know how to deserialize it

Does splunk have any UX/tools to parse a key value pair where the value has commas in it?

**Reply by randmonkey** [reply]
> > we could serialize it into a comma separated values like to: version=1.23,1.24
> 
> > This would make it non-trivial to parse, the consumer of this report would have to know how to deserialize it
> 
> Does splunk have any UX/tools to parse a key value pair where the value has commas in it?

also have the same question on results of mesh detection, the result would be `mesh-deploy=istio1,istio2`.
